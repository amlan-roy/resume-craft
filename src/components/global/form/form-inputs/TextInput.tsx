import React from "react";
import { Control, FieldValues } from "react-hook-form";
import {
  FormControl,
  FormField,
  FormItem,
  FormLabel,
  FormMessage,
} from "@/components/ui/form";
import { Input } from "@/components/ui/input";
import { Textarea } from "@/components/ui/textarea";

type TextInputProps = {
  fieldName: string;
  label?: string;
  placeholder?: string;
  multiline?: boolean;
  control?: Control<FieldValues> | undefined;
  type?: React.HTMLInputTypeAttribute;
  className?: string;
  autoComplete?: string;
  inputClassName?: string;
};

const TextInput: React.FC<TextInputProps> = ({
  control,
  fieldName,
  label,
  placeholder,
  type,
  multiline,
  className,
  inputClassName,
  autoComplete,
}) => {
  return (
    <FormField
      control={control}
      name={fieldName}
      render={({ field }) => (
        <FormItem className={className}>
          {label && (
            <FormLabel className="text-lg md:text-xl text-brand-neutral-11">
              {label}
            </FormLabel>
          )}
          <FormControl>
            {multiline ? (
              <Textarea
                placeholder={placeholder}
                {...field}
                autoComplete={autoComplete}
                className={inputClassName}
              />
            ) : (
              <Input
                placeholder={placeholder}
                {...field}
                type={type}
                autoComplete={autoComplete}
                className={inputClassName}
              />
            )}
          </FormControl>
          <FormMessage />
        </FormItem>
      )}
    />
  );
};
export default TextInput;
