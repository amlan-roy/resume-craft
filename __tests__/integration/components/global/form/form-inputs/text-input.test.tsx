import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import TextInput from "@/components/global/form/form-inputs/TextInput";

describe("integration:components/global/form/form-inputs/text-input", () => {
  it("snapshot", () => {
    const tree = render(
      <TextInput label="Name" placeholder="Enter your name" />
    );

    expect(tree.container).toMatchSnapshot();
  });
  it("renders a label and an input field", () => {
    render(<TextInput label="Name" placeholder="Enter your name" />);

    const label = screen.getByText("Name");
    const input = screen.getByPlaceholderText("Enter your name");

    expect(label).toBeInTheDocument();
    expect(input).toBeInTheDocument();
  });

  it("renders a textarea for multiline inputs", () => {
    render(
      <TextInput
        multiline
        label="Description"
        placeholder="Enter a description"
      />
    );

    const textarea = screen.getByRole("textbox"); // Use role for accessibility

    expect(textarea).toBeInTheDocument();
  });

  it("renders an error message if provided", () => {
    render(<TextInput errorMessage="Invalid input" />);

    const errorMessage = screen.getByText("Invalid input");

    expect(errorMessage).toBeInTheDocument();
  });

  it("applies custom className to the container", () => {
    render(<TextInput className="custom-class" />);

    const container = screen.getByTestId("text-input__container"); // Assuming you have a test-id

    expect(container).toHaveClass("custom-class");
  });
});
