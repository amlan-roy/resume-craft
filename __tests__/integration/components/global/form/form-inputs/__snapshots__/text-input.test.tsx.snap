// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration:components/global/form/form-inputs/text-input snapshot 1`] = `
<div>
  <div
    data-testid="text-input__container"
  >
    <label
      class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-lg md:text-xl text-brand-neutral-11"
    >
      Name
    </label>
    <input
      class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 my-3"
      placeholder="Enter your name"
    />
  </div>
</div>
`;
